### PR TITLE
feat: [FW-52] create generate page with the different steps

### PR DESCRIPTION
**Pull request by barroro**
- Add shadcn components needed to make the different requirements in each step.
- Create `Stepper` component
- Create custom Combobox for the search artists selector
- Add new tests for the generate playlist page
- Add new translations for the generate playlist page

### CODE REVIEW

File: locales/ca/generate.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+   "meta": {
+      "description": "Genera llistes de música a Spotify basat en les cançons que estàn tocant en aquell tour, utilizant la API de Setlist.fm",
```
Comment (DanielMoraDC) [inline locales/ca/generate.json:3]:
```suggestion
      "description": "Genera llistes de música a Spotify basat en les cançons que estan tocant en aquell tour, utilizant la API de Setlist.fm",
```

Comment (DanielMoraDC) [reply locales/ca/generate.json:3]: I also feel that this sentence feels a bit confusing. "que estan tocant" does not have a subject in the sentence. What about something like:

"Genera llistes amb les cançons que les teves bandes estan girant actualment"

Note that I am excluding de implementation details (e.g. Spotify, Setlistfm).

File: locales/ca/generate.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+         "generate": "Generar",
+         "next": "Següent",
+         "previous": "Anterior"
+      },
+      "step1": {
+         "description": "Selecciona per crear una nova llista o utilitza una llista ja existent en el teu compte",
+         "form": {
+            "createNewPlaylist": {
+               "description": "Utilitza aquesta opció per crear una nova llista de reproducció a Spotify",
+               "giveAName": "Dóna-li un nom a la llista",
```
Comment (DanielMoraDC) [inline locales/ca/generate.json:19]: I believe "li" here is unnecessary, as you are already including the list in the sentence, so it is not replacing anything as a pronoun, in the end.

File: locales/ca/generate.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+               "namePlaceholder": "Nom de la llista",
+               "privatePlaylist": {
+                  "description": "La llista de reproducció serà privada i només tu podràs veure-la",
+                  "title": "Llista privada"
+               },
+               "title": "Crear una nova llista"
+            },
+            "useExistingPlaylist": {
+               "description": "Utilitza aquesta opció per afegir cançons a una llista de reproducció ja existent",
```
Comment (DanielMoraDC) [inline locales/ca/generate.json:28]: We could add here that only lists owned by the user can be used.

File: locales/ca/generate.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+               "selectPlaylist": "Selecciona una llista de reproducció",
+               "title": "Utilitzar una llista existent"
+            }
+         },
+         "title": "Tria una opció per començar"
+      },
+      "step2": {
+         "description": "Busca els artistes que vols afegir a la llista de reproducció",
```
Comment (DanielMoraDC) [inline locales/ca/generate.json:36]: I would replace "busca" with "afegeix" in this second step. The ultimate goal is to add artists, searching is just the means to it.

File: src/components/generate/PlaylistGetUrlLink/PlaylistGetUrlLink.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { Button } from "@components/ui/Button"
```
Comment (DanielMoraDC) [inline src/components/generate/PlaylistGetUrlLink/PlaylistGetUrlLink.tsx:1]: On playlist url copy, I am missing some effect to acknowledge the copy operation (dialog, pop-up, etc). Could we add that?

**Comment by DanielMoraDC** [comment]
I left some comments.

We should use smaller PRs to ease the review/merge process. Smaller PRs are benefitial in many aspects (e.g. review quality, time, atomicity of changes). I am not blocking this one, as we agreed internally, but I will do for subsequent large PRs 🕵️

**Comment by DanielMoraDC** [comment]
The interface looks great, very good job 🚀

File: src/components/generate/PlaylistGetUrlLink/PlaylistGetUrlLink.tsx (hunk 5 above)
Comment (barroro) [reply src/components/generate/PlaylistGetUrlLink/PlaylistGetUrlLink.tsx:1]: This effect has been implemented in this task [FW-55](https://festwrap.atlassian.net/browse/FW-55)

File: src/components/generate/GeneratePlaylistStepper.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
+import PlaylistGetUrlLink from "@components/generate/PlaylistGetUrlLink/PlaylistGetUrlLink"
+import PlaylistSearchBandsForm from "@components/generate/PlaylistSearchBandsForm/PlaylistSearchBandsForm"
+import PlaylistSetupForm from "@components/generate/PlaylistSetupForm/PlaylistSetupForm"
+import { Button } from "@components/ui/Button"
+import { Stepper, StepList, Step, StepContent } from "@components/ui/Stepper"
+import useTranslation from "next-translate/useTranslation"
+import Link from "next/link"
+import { useState } from "react"
+
+const STEPS_COUNT = 3
```
[the remark this answers is not in the document]
Comment (barroro) [reply src/components/generate/GeneratePlaylistStepper.tsx:10]: I will check and create a task to refactor the `Stepper` component.

File: locales/ca/generate.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
+{
+   "meta": {
+      "description": "Genera llistes amb les cançons que les teves bandes estan girant actualment",
```
Comment (DanielMoraDC) [inline locales/ca/generate.json:3]:
```suggestion
      "description": "Genera llistes amb les cançons amb què les teves bandes preferides estan girant actualment",
```

File: locales/ca/generate.json [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+         "finish": "Finalitzar",
+         "generate": "Generar",
+         "next": "Següent",
+         "previous": "Anterior"
+      },
+      "step1": {
+         "description": "Crea una llista nova o utilitza una llista propia existent",
+         "form": {
+            "createNewPlaylist": {
+               "description": "Utilitza aquesta opció per crear una nova llista de reproducció a Spotify",
```
Comment (DanielMoraDC) [inline locales/ca/generate.json:18]: I believe that descriptions are quite redundant for both options: they do not provide additional information but just clutter the space. Should we remove them?

File: locales/ca/generate.json [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+               },
+               "title": "Crear una nova llista"
+            },
+            "useExistingPlaylist": {
+               "description": "Utilitza aquesta opció per afegir cançons a una de les teves llistes de reproducció ja existent",
+               "selectPlaylist": "Selecciona una llista de reproducció",
+               "title": "Utilitzar una llista existent"
+            }
+         },
+         "title": "Tria una opció per començar"
```
Comment (DanielMoraDC) [inline locales/ca/generate.json:33]: I would change this to be "Configura la llista" or similar. It is more specific.

File: locales/ca/generate.json [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      },
+      "step2": {
+         "description": "Selecciona els artistes que vols afegir a la llista de reproducció",
+         "emptyState": {
+            "description": "Comença a buscar artistes per afegir-los a la llista de reproducció",
+            "title": "No hi ha cap artista afegit"
+         },
+         "searchPlaceholder": "Cerca artistes",
+         "title": "Cercar els artistes"
```
Comment (DanielMoraDC) [inline locales/ca/generate.json:42]: We probably want to use the imperative in both step 2 and 3.

**Comment by DanielMoraDC** [comment]
Captura de pantalla 2024-12-28 a les 11 43 17

I have some comments:
- Does it make sense to remove the top margin of the left menu so it is aligned with the main layout?
- I see that the mobile version uses the same navigation menu, but places it at the top. Is this something we plan to change afterwards?
- When going back in the generation process, previous input information is not recovered. I understand this has not been implemented yet, right?

**Review by DanielMoraDC — APPROVED** [review]
I left some new comments, but approving anyway. I do not need all of them to be resolved at this point. As long as we can link them to the corresponding JIRA task, I am happy to resolve them (feel free to resolve them).

**Pull request merged**